actions/profile/edit.php

```diff
@@ -42,23 +42,27 @@ function profile_array_decoder(&$v) {
 		$value = _elgg_html_decode($value);
 	}
 
+	// convert tags fields to array values
+	if ($valuetype == 'tags') {
+		$value = string_to_tag_array($value);
+	}
+	
 	// limit to reasonable sizes
-	// @todo - throwing away changes due to this is dumb!
-	// ^^ This is a sticky form so changes aren't lost...?
-	if (!is_array($value) && $valuetype != 'longtext' && elgg_strlen($value) > 250) {
-		$error = elgg_echo('profile:field_too_long', array(elgg_echo("profile:{$shortname}")));
-		register_error($error);
-		forward(REFERER);
+	if ($valuetype != 'longtext') {
+		$check_values = (array) $value;
+		// also check tags/checkboxes/etc
+		foreach ($check_values as $v) {
```

Comment:
array_walk_recursive() and an anonymous function would work. Though I don't know why we do this for tags at all.